packages/medusa/src/api/routes/admin/discounts/create-dynamic-code.ts

```diff
@@ -24,24 +32,33 @@ import { MedusaError, Validator } from "medusa-core-utils"
 export default async (req, res) => {
   const { discount_id } = req.params
 
-  const schema = Validator.object().keys({
-    code: Validator.string().required(),
-    usage_limit: Validator.number().default(1),
-    metadata: Validator.object().optional(),
-  })
-
-  const { value, error } = schema.validate(req.body)
-
-  if (error) {
-    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
-  }
+  const validated = await validator(
+    AdminPostDiscountsDiscountDynamicCodesReq,
+    req.body
+  )
 
-  const discountService = req.scope.resolve("discountService")
-  const created = await discountService.createDynamicCode(discount_id, value)
+  const discountService: DiscountService = req.scope.resolve("discountService")
+  const created = await discountService.createDynamicCode(
+    discount_id,
+    validated
+  )
 
   const discount = await discountService.retrieve(created.id, {
     relations: ["rule", "rule.valid_for", "regions"],
   })
 
   res.status(200).json({ discount })
 }
+
+export class AdminPostDiscountsDiscountDynamicCodesReq {
+  @IsString()
+  @IsNotEmpty()
+  code: string
+
+  @IsNumber()
+  usage_limit = 1
```

Comment:
Not sure if this is needed as per our latest chat
```suggestion
  @IsNumber()
  @Type(() => Number)
  usage_limit = 1
```